src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -38,38 +57,88 @@ public PngEncoder() {
                 null);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code bufferedImage}.
+     * The new PngEncoder will use the provided {@code bufferedImage}.
+     *
+     * @param bufferedImage input image
+     * @return a new PngEncoder
+     */
     public PngEncoder withBufferedImage(BufferedImage bufferedImage) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code compressionLevel}.
+     * The new PngEncoder will use the provided {@code compressionLevel}.
+     *
+     * @param compressionLevel input image (must be between -1 and 9 inclusive)
+     * @return a new PngEncoder
+     */
     public PngEncoder withCompressionLevel(int compressionLevel) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code multiThreadedCompressionEnabled}.
+     * The new PngEncoder will use the provided {@code multiThreadedCompressionEnabled}.
+     *
+     * @param multiThreadedCompressionEnabled when {@code true}, multithreaded compression will be used
+     * @return a new PngEncoder
+     */
     public PngEncoder withMultiThreadedCompressionEnabled(boolean multiThreadedCompressionEnabled) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code srgbRenderingIntent}.
+     * The new PngEncoder will use the provided {@code srgbRenderingIntent}.
```

Comment:
Since this is the method used to get an sRGB chunk added to the resulting PNG we could expand this to say:
```
The new PngEncoder will add an sRGB chunk to the encoded PNG and use the provided {@code srgbRenderingIntent}.
```